### ISSUE TITLE
Tilix scrollback lags at large resolutions (>1080p)

### ISSUE THREAD

**Issue by Lyude** [gnunn1/tilix#1173]
Recently I noticed that on my laptop along with another machine I've tried (both of which have 4K displays), scrolling through the tilix scroll back is far slower then it feels like it should be. Keep in mind this happens with a default tilix configuration as well, so unlimited scrollback is definitely not involved here. The easiest way I've been able to reproduce this is to just maximize the terminal on a 4k display and run a command like dmesg, then try scrolling up and down. Compared to gnome-terminal, there is significantly more lag even with the same amount of content in both terminals. As well, sysprof seems to indicate most of the CPU usage is coming from libpixman, although I'm not sure if that's through calls that tilix is making, libvte, etc.

**Comment by gnunn1** [comment]
I'm using a QHD monitor (2560 x 1440) and no issues here but obviously this is much less resolution then 4K. A couple of questions as follows:

a. What video card are you using?

b. If you disable overlay scrollbars in Appearances does that fix the issue?

c. What is the output of ```tilix --version```

**Comment by Lyude** [comment]
a. Kabylake GPU, HD 620 8086:5916. I work on mesa/i915/i965 so I can say I'm mostly sure this isn't a GPU issue (tilix's CPU usage rises to 100% when the lag happens), but if you want me to double check at some point let me know. Honestly the far more worrying I see is the fact that it's relying so heavily on pixman, anything pixman does is going to be 100% software acceleration which would also explain why it seems to lag so much more with larger scrollbacks
b. Yes! Now I can use my terminal again without heating up my laptop 👍 
c.
```
➜  ~ tilix --version
Versions
	Tilix version: 1.6.4
	VTE version: 0.48
	GTK Version: 3.22.21

Tilix Special Features
	Notifications enabled=1
	Triggers enabled=0
	Badges enabled=0
```

Also realized I completely forgot to mention that this is happening on Fedora 26.

**Comment by gnunn1** [comment]
I think this is a bug with either VTE or GTK with the overlay scrollbar, I'm tending to think VTE though. I'm pushing the envelope a bit by using the overlay scrollbar which is why there is an option to disable it.

Is anyone else having this issue?

**Comment by MingcongBai** [comment]
I am also having this issue with my Chromebook Pixel 2 - 2560*1700 screen. Tilix is quite slow whenever screen contents fills beyond the window size - requiring scrolling, when using xf86-video-intel; this issue is non-existence with Modesetting DDX.

It doesn't matter whether overlay scrollbar was on or not.

**Comment by MingcongBai** [comment]
Additionally, other VTE (GTK+ 3) based terminal emulators like GNOME Terminal exhibited no such issue.

**Comment by MingcongBai** [comment]
Hmm. It seems like if the window is maximised the issue is gone.

**Comment by TraceyC77** [comment]
I've just come across this bug report while searching for symptoms similar to what I was just seeing. I also saw a performance issue as Lyude described in the original post. I have 2 monitors which are 1920 x 1200 (so not 4k)

I observed unusually high CPU usage with tilix (40 - 90%) with the overlay scrollbar enabled
(Preferences: Appearance - Overlay scrollbar)
Tilix had been started just after a system reboot this morning. Performance was more and more sluggish as time went on. Also, when running "top", the output would flip between the normal one line output you'd expect and having a space in between each line.

After disabling the overlay scrollbar and restarting tilix it is now performing as quickly as I'd expect vs other open applications. It is no longer showing in the top of the output of "top". The line spacing issue is no longer present.

**System information:**
Linux Mint 18.3 Cinnamon

% tilix --version
	Tilix version: 1.6.4
	VTE version: 0.44
	GTK Version: 3.22.11

Tilix Special Features
	Notifications enabled=0
	Triggers enabled=0
	Badges enabled=0
